File: 217-contains-duplicate/contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # #Approach 1
        # for index in range(len(nums)):
        #     for checker_index in range(len(nums)):
        #         if(index == checker_index):
        #             continue
        #         if nums[index] == nums[checker_index]:
        #             return True
            
        # return False

        #Approach 2

        nums.sort()
        print(nums)
        for index in range(len(nums)):
            if index == 0:
                continue
            if nums[index] == nums[index - 1]:
                return True
        
        return False

File: 217-contains-duplicate/contains-duplicate.ts
function containsDuplicate(nums: number[]): boolean {
    // //1
    // let store: Set<number> = new Set()
    // let i:number = 0
    // while(i < nums.length){
    //     if(store.has(nums[i])){
    //         return true
    //     }
    //     store.add(nums[i])
    //     i+=1
    // }
    // return false

    if (nums.length == 1){
        return false
    }

    nums.sort()

    let i: number | String = 0
    let j: number | boolean = 1
    while(j< nums.length){
        if (nums[i] === nums[j]){
            return true
        }
        i+=1
        j+=1
    }
    return false
};